### PR TITLE
Varvara/feature/send sms aws

### PR DESCRIPTION
**Pull request by varvarashcherbyna7**

### CODE REVIEW

File: api/controllers/registration.py [hunk 1]
```diff
@@ -1,59 +1,120 @@
...
+            user.locations.append(location)
+
+        db.commit()
+        db.refresh(user)
+
+        log(log.INFO, "User [%s] was created", user.fullname)
+
+        # generate otp code
+        otp_code = random.randint(100000, 999999)
+        print("otp_code", otp_code)
```
Comment (Taraslysak) [inline api/controllers/registration.py]: Redundant print

File: api/controllers/registration.py [hunk 2]
```diff
@@ -1,59 +1,120 @@
...
 from sqlalchemy.orm import Session
 from sqlalchemy.sql.expression import Executable
+from mypy_boto3_sns import SNSClient
 
 from app import models as m
 from app import schema as s
 from app.logger import log
 
-from .oauth2 import create_access_token
+# from .oauth2 import create_access_token
```
Comment (Taraslysak) [inline api/controllers/registration.py]: Redundant comment

File: api/routes/registration.py [hunk 3]
```diff
@@ -15,16 +16,43 @@
...
+    },
+)
+def phone_verification(
+    phone_data: s.PhoneVerificationIn,
+    db=Depends(get_db),
+):
+    """Logs in a user, returns access token"""
+    log(log.INFO, "Phone verification for user with phone [%s]", phone_data.phone)
+
+    return c.phone_verification(phone_data, db)
```
Comment (Taraslysak) [inline api/routes/registration.py]: Function names should start from verbs

c.verify_phone

File: api/routes/registration.py [hunk 4]
```diff
@@ -35,7 +63,11 @@ def register_user(auth_data: s.RegistrationIn, db=Depends(get_db)):
     },
     response_model=s.SetPhoneOut,
 )
-def set_phone(phone_data: s.SetPhoneIn, db=Depends(get_db), current_user: m.User = Depends(get_current_user)):
+def set_phone(
+    phone_data: s.SetPhoneIn,
```
Comment (Taraslysak) [inline api/routes/registration.py:186]: We allow users to change their phone numbers?

File: api/controllers/registration.py [hunk 5]
```diff
@@ -1,59 +1,120 @@
...
+    if not user:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found")
+
+    if not user.otp_code:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Phone validation failed")
+
+    if user.otp_code != phone_data.otp_code:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid OTP code")
+
+    user.phone_verified = True
```
Comment (Taraslysak) [inline api/controllers/registration.py:114]: Please, clear verification code for user, so those code cannot be used second time

**Review by Taraslysak — COMMENTED** [review]
Good job!

Please consider some refactoring. Take close attention on clearing otp code after verification

File: api/routes/registration.py (hunk 4 above)
Comment (varvarashcherbyna7) [reply api/routes/registration.py:186]: it used to be functional, the registration stage, that is, user is already on the page where he need to enter the code, he cannot return to the previous one, but he can change the phone number, for example, he entered a number incorrectly and the code does not come

**Pull request merged**